File: backend/src/main/kotlin/com/trackwatch/backend/utils/helper/ImageHelper.kt
package com.trackwatch.backend.utils.helper

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient

@Component
class ImageHelper() {

    private val webClient = WebClient.builder().build()
    private val log = LoggerFactory.getLogger(ImageHelper::class.java)

    fun encodeImageToBase64(imageUrl: String): String? {
        return try {
            val bytes = webClient.get()
                .uri(imageUrl)
                .retrieve()
                .bodyToMono(ByteArray::class.java)
                .block()

            bytes?.let { java.util.Base64.getEncoder().encodeToString(it) }

        } catch (e: Exception) {
            log.error("Error while encoding image to base64", e)
            null
        }
    }

}

File: backend/app/utils/image_helper.py
import requests
import base64

def encode_image_to_base64(image_url: str) -> str:
  try:
    response = requests.get(image_url, timeout=10)
    response.raise_for_status()
    encoded = base64.b64encode(response.content).decode("utf-8")
    return encoded

  except Exception as e:
    print(f"Error while encoding image to base64: {str(e)}")
    return None
